File: src/413. Arithmetic Slices/main.go
// Solved by @kitanoyoru
// https://leetcode.com/problems/arithmetic-slices

func numberOfArithmeticSlices(nums []int) int {
  if len(nums) == 0 {
    return 0
  }

  diff, ans := 0, 0

  for i := 0; i < len(nums) - 2; i++ {
    diff = nums[i+1] - nums[i]
    for j := i + 2; j < len(nums); j++ {
      if nums[j] - nums[j-1] == diff {
        ans++
      } else {
        break
      }
    }
  }

  return ans 
}

File: src/413. Arithmetic Slices/main.py
# Solved by @kitanoyoru
# https://leetcode.com/problems/arithmetic-slices/


class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        if not len(nums):
            return 0

        diff, ans = 0, 0

        for i in range(0, len(nums) - 2):
            diff = nums[i + 1] - nums[i]
            for j in range(i + 2, len(nums)):
                if nums[j] - nums[j - 1] == diff:
                    ans += 1
                else:
                    break

        return ans
